### ISSUE TITLE
fetchAuthSession  has error that session expired.

### ISSUE THREAD

**Issue by hysuki** [aws-amplify/amplify-flutter#409]
**Describe the bug**
when I call fetchAuthSession function for taking access token, has an error that session has expired.

1. successful sign in(with Facebook, Google(web UI), cognito userpool(auth.SignUp, SignIn))
2. call fetchAuthSession function
3. a 'SIGNED OUT' event occurs in the event hub.
4. print error that 'has expired session'.

**Expected behavior**
can take accessToken

**Screenshots**
![sessionExpired2](https://user-images.githubusercontent.com/35303552/109205918-97564c80-77ea-11eb-8491-5f0d4ba4c8d5.gif)

**Additional context**
amplify_flutter: '<1.0.0'
amplify_auth_cognito: '<1.0.0'

**Comment by haverchuck** [comment]
@hysuki Which specific version of amplify_flutter and amplify_auth_cognito are you currently using?

**Comment by hysuki** [comment]
@haverchuck 

### package version

amplify_flutter: ^0.1.0
amplify_auth_cognito: ^0.1.0
---
### code
```dart 
 Future<void> signInWithGoogle() async {
    try {
      SignInResult res =
          await Amplify.Auth.signInWithWebUI(provider: AuthProvider.google);
      logger.d(res);
      await fetchSession();
    } on AuthException catch (e) {
      logger.e(e.message);
    }
  }

Future<void> fetchSession() async {
    try {
      CognitoAuthSession res = await Amplify.Auth.fetchAuthSession(
          options: CognitoSessionOptions(getAWSCredentials: true)
      );
      logger.d("accessToken : " + res.userPoolTokens.accessToken);
    } on AuthException catch (e) {
      logger.e(e.message);
    }
  }
```

### log
```bash
W/InternalCallback(20351): Library attempted to call user callback twice, expected only once
W/InternalCallback(20351): Library attempted to call user callback twice, expected only once
D/AWSMobileClient(20351): Inspecting user state details
D/AWSMobileClient(20351): Inspecting user state details
D/AWSMobileClient(20351): waitForSignIn: userState:SIGNED_OUT
I/flutter (20351): ┌───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
I/flutter (20351): │ #0   AuthController.authHub.<anonymous closure> (package:airwith/src/controllers/auth_controller.dart:37:20)
I/flutter (20351): │ #1   _rootRunUnary (dart:async/zone.dart:1198:47)
I/flutter (20351): ├┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄
I/flutter (20351): │ 🐛 USER IS SIGNED OUT
I/flutter (20351): └───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
E/amplify:flutter:auth_cognito(20351): SessionExpiredException
E/amplify:flutter:auth_cognito(20351): SessionExpiredException{message=Your session has expired., cause=null, recoverySuggestion=Please sign in and reattempt the operation.}
E/amplify:flutter:auth_cognito(20351): 	at com.amazonaws.amplify.amplify_auth_cognito.AuthCognito.prepareCognitoSessionFailure(AuthCognito.kt:478)
E/amplify:flutter:auth_cognito(20351): 	at com.amazonaws.amplify.amplify_auth_cognito.AuthCognito$onFetchAuthSession$1.accept(AuthCognito.kt:331)
E/amplify:flutter:auth_cognito(20351): 	at com.amazonaws.amplify.amplify_auth_cognito.AuthCognito$onFetchAuthSession$1.accept(AuthCognito.kt:68)
E/amplify:flutter:auth_cognito(20351): 	at com.amplifyframework.auth.cognito.MobileClientSessionAdapter$1.onError(MobileClientSessionAdapter.java:74)
E/amplify:flutter:auth_cognito(20351): 	at com.amazonaws.mobile.client.internal.ReturningRunnable$1.run(ReturningRunnable.java:47)
E/amplify:flutter:auth_cognito(20351): 	at java.lang.Thread.run(Thread.java:923)
I/flutter (20351): ┌───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
I/flutter (20351): │ #0   AuthController.fetchSession (package:airwith/src/controllers/auth_controller.dart:72:14)
I/flutter (20351): │ #1   <asynchronous suspension>
I/flutter (20351): ├┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄
I/flutter (20351): │ ⛔ Your session has expired.
I/flutter (20351): └───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
D/AWSMobileClient(20351): Inspecting user state details
D/AWSMobileClient(20351): waitForSignIn: userState:SIGNED_OUT

```

**Comment by haverchuck** [comment]
@hysuki Thanks for the additional information.  The code you've provided is working for me. It's curious that the session failure seems to be precipitated by a signout event.  Did you set up with your Cognito resources with the Amplify CLI?  If so, did you use the default options, or did you change anything?  Feel free to post your amplifyconfiguration.dart with any sensitive values obscured.

**Comment by hysuki** [comment]
@haverchuck I was set Cognito resource by using @aws-amplify/cli@4.44.1.
that include user pool group, social login(facebook, google) and custom message, post confirm triggers(add to group, custom).

amplifyconfiguration.dart 
```dart
const amplifyconfig = ''' {
    "UserAgent": "aws-amplify-cli/2.0",
    "Version": "1.0",
    "auth": {
        "plugins": {
            "awsCognitoAuthPlugin": {
                "UserAgent": "aws-amplify-cli/0.1.0",
                "Version": "0.1.0",
                "IdentityManager": {
                    "Default": {}
                },
                "CredentialsProvider": {
                    "CognitoIdentity": {
                        "Default": {
                            "PoolId": "us-east-1:xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
                            "Region": "us-east-1"
                        }
                    }
                },
                "CognitoUserPool": {
                    "Default": {
                        "PoolId": "us-east-1_xxxxxxxxx",
                        "AppClientId": "xxxxxxxxxxxxxxxxxxxxxxxxxx",
                        "Region": "us-east-1"
                    }
                },
                "Auth": {
                    "Default": {
                        "OAuth": {
                            "WebDomain": "xxxxxx-dev.auth.us-east-1.amazoncognito.com",
                            "AppClientId": "xxxxxxxxxxxxxxxxxxxxxxxxxx",
                            "SignInRedirectURI": "myapp://callback/",
                            "SignOutRedirectURI": "myapp://signout/",
                            "Scopes": [
                                "phone",
                                "email",
                                "openid",
                                "profile",
                                "aws.cognito.signin.user.admin"
                            ]
                        },
                        "authenticationFlowType": "USER_SRP_AUTH"
                    }
                },
                "AppSync": {
                    "Default": {
                        "ApiUrl": "https://xxxxxxxxxxxxxxxxxxxxxxxxxx.appsync-api.us-east-1.amazonaws.com/graphql",
                        "Region": "us-east-1",
                        "AuthMode": "API_KEY",
                        "ApiKey": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
                        "ClientDatabasePrefix": "airwith_API_KEY"
                    },
                    "airwith_AMAZON_COGNITO_USER_POOLS": {
                        "ApiUrl": "https://xxxxxxxxxxxxxxxxxxxxxxxxxx.appsync-api.us-east-1.amazonaws.com/graphql",
                        "Region": "us-east-1",
                        "AuthMode": "AMAZON_COGNITO_USER_POOLS",
                        "ClientDatabasePrefix": "airwith_AMAZON_COGNITO_USER_POOLS"
                    }
                },
                "PinpointAnalytics": {
                    "Default": {
                        "AppId": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
                        "Region": "us-east-1"
                    }
                },
                "PinpointTargeting": {
                    "Default": {
                        "Region": "us-east-1"
                    }
                },
                "S3TransferUtility": {
                    "Default": {
                        "Bucket": "xxxxxxxx-xxxxxxxxxxxxx-dev",
                        "Region": "us-east-1"
                    }
                }
            }
        }
    },
    "analytics": {
        "plugins": {
            "awsPinpointAnalyticsPlugin": {
                "pinpointAnalytics": {
                    "appId": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
                    "region": "us-east-1"
                },
                "pinpointTargeting": {
                    "region": "us-east-1"
                }
            }
        }
    },
    "api": {
        "plugins": {
            "awsAPIPlugin": {
                "airwith": {
                    "endpointType": "GraphQL",
                    "endpoint": "https://xxxxxxxxxxxxxxxxxxxxxxxxx.appsync-api.us-east-1.amazonaws.com/graphql",
                    "region": "us-east-1",
                    "authorizationType": "API_KEY",
                    "apiKey": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
                }
            }
        }
    },
    "storage": {
        "plugins": {
            "awsS3StoragePlugin": {
                "bucket": "xxxxxxxx-xxxxxxxxxxxxxx-dev",
                "region": "us-east-1",
                "defaultAccessLevel": "guest"
            }
        }
    }
}''';
```

**Comment by hysuki** [comment]
IOS have a different error message from android in the same code

```bash
flutter: ┌───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
flutter: │ #0   AuthController.signInWithGoogle (package:airwith/src/controllers/auth_controller.dart:86:14)
flutter: │ #1   <asynchronous suspension>
flutter: ├┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄
flutter: │ 🐛 Instance of 'CognitoSignInResult'
flutter: └───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
flutter: ┌───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
flutter: │ #0   AuthController.authHub.<anonymous closure> (package:airwith/src/controllers/auth_controller.dart:28:20)
flutter: │ #1   _rootRunUnary (dart:async/zone.dart:1198:47)
flutter: ├┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄
flutter: │ 🐛 USER IS SIGNED IN
flutter: └───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
flutter: [38;5;196m┌───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────[0m
flutter: [38;5;196m│ #0   AuthController.fetchSession (package:airwith/src/controllers/auth_controller.dart:68:14)
flutter: [38;5;196m│ #1   <asynchronous suspension>
flutter: [38;5;196m├┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄[0m
flutter: [38;5;196m│ ⛔ Unexpected error occurred with message: An unknown error occurred[0m
flutter: [38;5;196m└───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────[0m
```

```bash
error = {
	message = Unexpected error occurred with message: An unknown error occurred
	recoverySuggestion = This should not happen. There is a possibility that there is a bug if this error persists. Please take a look at https://github.com/aws-amplify/amplify-ios/issues to see if there are any existing issues that match your scenario, and file an issue with the details of the bug if there isn't. Issue encountered at:
file: /Users/code/project/sync/application/ios/Pods/Amplify/Amplify/Categories/Auth/Error/AuthError.swift
function: recoverySuggestion
line: 79
	underlyingException = The operation couldn’t be completed. (com.amazonaws.AWSCognitoIdentityErrorDomain error 10.)
}
```

**Comment by haverchuck** [comment]
@omchiii Thanks for this information. I'm going to try replicating your config as much as possible.

**Comment by hysuki** [comment]
@haverchuck  I used amplifyconfiguration, which is automatically generated by amplify, and I did not make any manual changes.

However, for that configuration, I used manual configuration, not default configuration option, and after that the first configuration, have updated several times for adding and removing triggers by using amplify update auth.

When updating, I used Walkthrough all the auth configurations option.

**Comment by haverchuck** [comment]
@hysuki Would you be interested in chatting about this on Discord or some other medium?

**Comment by hysuki** [comment]
@haverchuck 
I'd like to have a discord.

I checked my cognito identity after seeing your update, and I found out that identity pool is missing.

When I created a new identity pool for the existing project and connected it to the cognito user pool, I was able to confirm that I successfully imported accessToken.

To reproduce, we made a new sample project and tested it.

When i was created new auth and updating through amplify update auth i saw ths message that
```bash
Warning! Your existing IdentityPool: samplec328b17ac328b17a_identitypool_4749a147 will be deleted upon the next “amplify push”!
```

I understood that this content was to remove the existing identity pool and create a new one, but I checked that only the existing pool was removed.

I'm not sure if this is intended in cli or not, but it seems to be a problem with amplify cli, not with amplify-flutter.
I think update should regenerate identity pool.

**Comment by haverchuck** [comment]
@hysuki When is a good time to chat (and what is your timezone)?

**Comment by hysuki** [comment]
@haverchuck 
I have free time between 2:00 PM and 8:00 PM (UTC).

**Comment by haverchuck** [comment]
@hysuki Would 7:30 PM (UTC) work for you on Thursday, March 4?

**Comment by hysuki** [comment]
@haverchuck 
It's possible at that time.

**Comment by haverchuck** [comment]
@hysuki Here is our Discord: discord.gg/amplify
I am listed under "Amplify Staff" as "dustin".

**Comment by Sunsiha** [comment]
any update on this?

**Comment by mmiravetedc** [comment]
Same issue here when trying to `fetchAuthSession()`. Any idea of what might be happenning?

**Comment by fjnoyp** [comment]
Hi @mmiravetedc this issue is a bit old so the context information provided to us may be out of date.  Can you create a new issue with details on your issue (which platforms affected, versions used, how to reproduce)
